test/e2e_node/restart_test.go

```diff
@@ -208,5 +211,88 @@ var _ = SIGDescribe("Restart [Serial] [Slow] [Disruptive]", func() {
 				}
 			}
 		})
+
+		ginkgo.It("should restart successfully when PodSandboxes are missing", func() {
+			containerRuntimeEndpoint := framework.TestContext.ContainerRuntimeEndpoint
+			// We only run this test on containerd nodes for now - I'm not sure if we
+			// have a CRI independent way of removing containers outside of the kubelet api.
+			if !strings.HasSuffix(containerRuntimeEndpoint, "/containerd.sock") {
+				e2eskipper.Skipf("the container runtime is not containerd")
+			}
+			if _, err := exec.Command("which", "ctr").CombinedOutput(); err != nil {
+				e2eskipper.Skipf("'ctr' command is not found")
+			}
+			if _, err := exec.Command("which", "crictl").CombinedOutput(); err != nil {
+				e2eskipper.Skipf("'crictl' command is not found")
+			}
+
+			ginkgo.By("creating a pod on the node")
+			podCountRestartAlways := 1
+			pods := newTestPods(1, false, imageutils.GetE2EImage(imageutils.BusyBox), "restart-kubelet-test2")
+			for _, pod := range pods {
+				pod.Spec.Containers[0].Command = []string{"sleep", "3d"}
+			}
+			createBatchPodWithRateControl(f, pods, podCreationInterval)
+			defer deletePodsSync(f, pods)
+
+			completedPods := waitForPods(f, podCountRestartAlways, startTimeout)
+			if len(completedPods) < podCountRestartAlways {
+				framework.Failf("Failed to run sufficient pods, got '%d' but expected '%d'", len(completedPods), podCount)
+			}
+
+			ginkgo.By("removing a pod sandbox's '/pause' container")
+
+			// retrieve pods on the node
+			getPodsCmd := fmt.Sprintf("crictl --runtime-endpoint %q pods --quiet", containerRuntimeEndpoint)
+			stdout, err := exec.Command("sudo", "bash", "-c", getPodsCmd).CombinedOutput()
+			if err != nil {
+				framework.Failf("Failed to get pods using crictl command: %q, err: %v, stdout: %q", getPodsCmd, err, string(stdout))
+			}
+
+			// find a pod sandbox ID
+			var podSandboxID string
```

Comment:
This is janky, but it's ~fine. If we add more tests that need to account for Kubelet/CRI differences we'll need to write a wrapper around at least containerd+crio, but I'd rather us have any coverage than none.